tasks/rabbitmq_vhosts.yml

```diff
@@ -1,8 +1,27 @@
 ---
-- name: Create vhosts
+- name: rabbitmq_extra_vhosts | Create vhosts
   rabbitmq_vhost:
     name: "{{ item['name'] }}"
     state: "{{ item['state'] }}"
   with_items: "{{ rabbitmq_extra_vhosts }}"
   run_once: "{{ rabbitmq_enable_clustering is defined and rabbitmq_enable_clustering }}"
   delegate_to: "{{ rabbitmq_master|default(omit) }}"
+  register: rabbitmq_created_vhosts
+
+- name: rabbitmq_extra_vhosts | Check guest administrator is present
+  command: rabbitmqctl -q list_users
+  become: true
+  run_once: "{{ rabbitmq_enable_clustering is defined and rabbitmq_enable_clustering }}"
+  delegate_to: "{{ rabbitmq_master|default(omit) }}"
+  when: rabbitmq_created_vhosts.changed
```

Comment:
Can you add a changed_when: false to indcate it's read only ?

Reply:
:+1: